### PR TITLE
[webui] Do not set the default project for buildresults in the contro…

### PR DESCRIPTION
**Pull request by hennevogel**
…ller

Set it in the model method to avoid that developers accidently save the @package with a 'wrong' project.

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -878,8 +878,9 @@ def build_result_old(repository, view = [])
     Buildresult.find(project: project, package: self, repository: repository, view: view)
   end
 
-  def buildresults
-    results = Buildresult.find_hashed(project: project.name, package: name, view: 'status', multibuild: '1', locallink: '1')
+  def buildresults(prj = nil)
+    prj = project unless prj
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: def buildresults(prj = project)

;-)

**Comment by adrianschroeter** [comment]
On Montag, 6. Februar 2017, 23:30:37 CET wrote Björn Geuken:
> bgeuken commented on this pull request.
> 
> 
> 
> > @@ -878,8 +878,9 @@ def build_result_old(repository, view = [])
>      Buildresult.find(project: project, package: self, repository: repository, view: view)
>    end
>  
> -  def buildresults
> -    results = Buildresult.find_hashed(project: project.name, package: name, view: 'status', multibuild: '1', locallink: '1')
> +  def buildresults(prj = nil)
> +    prj = project unless prj
> 
> def buildresults(prj = project)

IMHO there shouldn't be a "= nil", because I can not imagine a usecase.
(In which situation it is not important in which project a package build? Does
 the same check exists in multiple places?)

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

File: src/api/app/models/package.rb (hunk 1 above)
Comment (Ana06) [inline src/api/app/models/package.rb]: Why? : confused:

The [Ruby style guide](https://github.com/bbatsov/ruby-style-guide/blob/master/README.md#spaces-around-equals) says:

> Use spaces around the `=` operator when assigning default values to method parameters


Although for example _pep8_, the style analizer for Python, likes just the opposite

Comment (bgeuken) [inline src/api/app/models/package.rb]: @Ana06 Could you explain what you mean?

Comment (bgeuken) [inline src/api/app/models/package.rb]: There are spaces around the = operator. My point was rather that if the default value for prj is self.project, then we could define that the method head already.

Comment (Ana06) [inline src/api/app/models/package.rb]: Ah, okey, I didn't see them, ignore me then :joy:

File: src/api/app/views/webui/package/_buildstatus.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<% @package.buildresults.each_pair do |package, results| %>
+<% @package.buildresults(@project).each_pair do |package, results| %>
```
Comment (evanrolfe) [inline src/api/app/views/webui/package/_buildstatus.html.erb]: Can we do this in a way that doesn't add more logic to the view? i.e. do this in the controller?

```ruby
@buildresults = @package.buildresults(@project)
```

**Comment by hennevogel** [comment]
@adrianschroeter @evanrolfe this is where we come from. A FIXME about Package.get_by_project_and_name. This is just another form of stupid workaround...

**Comment by evanrolfe** [comment]
@hennevogel I don't understand.. The FIXME comment says:

> if you call Package.get_by_project_and_name( 'blah', 'blubb', {follow_project_links: true} ) then @package.project might not be 'blah' but another project

But in this PR, instead of fixing `Package.get_by_project_and_name` directly you code a workaround for it instead? Why not just fix `Package.get_by_project_and_name` directly so we don't need a "stupid workaround"?

**Comment by adrianschroeter** [comment]
On Freitag, 10. Februar 2017, 01:31:21 CET wrote Evan Rolfe:
> @hennevogel I don't understand.. The FIXME comment says:
> 
> > if you call Package.get_by_project_and_name( 'blah', 'blubb', {follow_project_links: true} ) then @package.project might not be 'blah' but another project
> 
> But in this PR, instead of fixing `Package.get_by_project_and_name` directly you code a workaround for it instead? Why not just fix `Package.get_by_project_and_name` directly so we don't need a "stupid workaround"?

Package.get_by_project_and_name  is not broken, it does exactly what it has to do: providing
the source/meta object. And this is critical for permission checking for source/meta changes.

But since the results depend on a different object (the build object) it can't tell you in which
project the build results are. This a 1 to many relationship (or even 0 to many when you take remote 
package objects into accounts).

That was the reason why I argued last year that buildresults do not belong to Package class,
but more to something like the BuildContainer class which we had (but in this usecase without
a database/activerecord object).

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by evanrolfe** [comment]
Ah ok, from the FIXME comment it seemed like `Package.get_by_project_and_name` was broken.

I'm fine for this to be merged then.

**Comment by hennevogel** [comment]
rebased. @evanrolfe please review again

**Comment by bgeuken** [comment]
@hennevogel Looks like the last 2 commits just adjust the first one. Please squash them all together:-)

LGTM

**Pull request merged**